alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +170,70 @@ class AlertingSettings {
             -1L,
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
+
+        val METRICS_STORE_TIME = Setting.timeSetting(
+            METRICS_STORE_TIME_DEFAULT.key,
+            METRICS_STORE_TIME_DEFAULT,
+            StorageValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        val METRICS_EXECUTION_FREQUENCY = Setting.timeSetting(
+            METRICS_EXECUTION_FREQUENCY_DEFAULT.key,
+            METRICS_EXECUTION_FREQUENCY_DEFAULT,
+            ExecutionValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+        internal class ExecutionValidator : Setting.Validator<TimeValue> {
```

Comment:
You'll want to extract your validation logic into a separate, helper file. That will help prevent the `AlertingSettings` file from becoming clogged with feature-specific validators.